### PR TITLE
feat: LogCollector use alloy_primitives::Log

### PR DESCRIPTION
**Pull request by qiweiii**
## Motivation

resolve: #6746

### CODE REVIEW

**Comment by qiweiii** [comment]
hi @mattsse, are my changes the expected solution?

File: crates/forge/tests/it/repros.rs [hunk 1]
```diff
@@ -123,8 +126,14 @@ test_repro!(3347, false, None, |res| {
...
         anonymous: false,
     };
-    let raw_log =
-        RawLog { topics: test.logs[0].topics.clone(), data: test.logs[0].data.clone().to_vec() };
+    let raw_log = RawLog {
+        topics: test.logs[0]
+            .topics
+            .iter()
+            .map(|topic| H256::from_slice(topic.as_slice()))
+            .collect(),
```
Comment (mattsse) [inline crates/forge/tests/it/repros.rs:134]: this should use `to_ethers()` for now I believe

File: crates/evm/evm/src/inspectors/logs.rs [hunk 2]
```diff
@@ -67,8 +67,22 @@ fn convert_hh_log_to_event(call: HardhatConsole::HardhatConsoleCalls) -> Log {
...
+        address: Address::default(),
+        topics: Vec::default(),
+        data: Bytes::default(),
+        block_hash: None,
+        block_number: None,
+        transaction_hash: None,
+        transaction_index: None,
+        log_index: None,
+        removed: false,
     }
```
Comment (mattsse) [inline crates/evm/evm/src/inspectors/logs.rs:87]: this can be removed once I've merged this:

https://github.com/alloy-rs/alloy/pull/101

**Review by mattsse — CHANGES_REQUESTED** [review]
yep, this is great!

only smol nits

File: crates/evm/evm/src/inspectors/logs.rs (hunk 2 above)
Comment (mattsse) [inline crates/evm/evm/src/inspectors/logs.rs:87]: this can now be replace with the Default impl

**Review by mattsse — CHANGES_REQUESTED** [review]
this lgtm

pending Default usage and sanity check @DaniPopes

**Comment by DaniPopes** [comment]
Needs rebase to alloy-primitives 0.6.0. `Log` is now `LogData` among other things.

**Comment by qiweiii** [comment]
Does this mean I need to update `alloy_rpc_types::Log` to `use alloy_primitives::Log`

**Comment by DaniPopes** [comment]
Yes that would be ideal, since the only fields we're using are data and topics

**Comment by mattsse** [comment]
that sounds reasonable!

**Review by mattsse — APPROVED** [review]
lgtm,
pending @DaniPopes

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**